app/components/layout/sidebar/sidebar.tsx

```diff
@@ -17,6 +22,17 @@ export default function Sidebar() {
   const [isMobileNavOpen, toggleMobileNav] = useAtom(toggleMobileNavAtom);
   const mainNavigationRef = useRef<HTMLElement>(null);
 
+  const [showScanner, setShowScanner] = useState(false);
+
+  const { stopMediaStream } = useMediaStream();
+  useStopMediaStream(stopMediaStream);
+
+  const handleScannerClose = () => {
+    stopMediaStream();
+    setShowScanner(false);
+    window.location.reload(); // Clear error messages when closing the scanner
```

Comment:
I don't understand why we have to reload the page. Can you clarify what is going on here?

Reply:
if we don't reload then it's also fine, I have added that to make sure camera access get's terminated properly because on some browser packages don't support some features